chemreg/compound/serializers.py

```diff
@@ -99,6 +97,27 @@ def validate_inchikey(self, value):
             raise ValidationError("CID must be included when InchIKey is defined.")
         return value
 
+    def validate(self, data):
+        qs = self.Meta.model.objects.filter(inchikey=data["inchikey"])
+        if qs.exists() and not self.admin_override:
+            dom = get_current_site(self.context["request"]).domain
+            matched = []
+            for obj in qs:
+                ext = reverse("definedcompound-detail", kwargs={"pk": obj.pk})
+                matched.append(f"http://{dom}{ext}")
```

Comment:
DRF has a reverse method that won't fail on the back end if the request is not set (interactions in tests and shell).

```python
        if qs.exists() and not self.admin_override:
            request = self.context.get("request", None)
            matched = []
            for obj in qs:
                import from rest_framework.reverse import reverse as drf_reverse
                matched.append(
                    drf_reverse("definedcompound-detail", request=request, kwargs={"pk": obj.pk})
                )
```

Reply:
cool, looks like if `request` comes out to be `None` it (`drf_reverse`) won't fail
```
(Pdb) request
(Pdb) drf_reverse("definedcompound-detail", request=request, kwargs={"pk": obj.pk})
'/definedCompounds/717
```
so we don't need to handle that case. I like it!